app/controllers/auth_source_ldaps_controller.rb

```diff
@@ -18,6 +20,23 @@ def create
     end
   end
 
+  def test_connection
+    result = {}
+    begin
+      Timeout::timeout(10) do
+        temp_auth_source_ldap = AuthSourceLdap.new(params[:auth_source_ldap])
+        temp_auth_source_ldap.ldap_con.valid_user?("")
+      end
+      result[:success] = true
+      result[:message] = _("Connection to LDAP Server Successful !!")
+    rescue => exception
```

Comment:
Maybe just catch `Net::LDAP::Error` and the timeout exception here, just in case we do too much?